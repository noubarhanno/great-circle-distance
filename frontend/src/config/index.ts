export const BASE_URL = "http://localhost:4000/api";

export const endpoints = {
  GET_NEARBY_PARTNERS: "/partners/get-nearby-partners",
};
