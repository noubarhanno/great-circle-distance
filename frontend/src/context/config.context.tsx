import React, { createContext, useContext, useReducer } from "react";
import { EThemeMode } from "../interfaces";

/**
 * config reducer actions
 */
export enum EConfigContextAction {
  SET_THEME_MODE = "SET_THEME_MODE",
  SET_LOCALE = "SET_LOCALE",
  SET_ERROR = "SET_ERROR",
  CLEAR_ERROR = "CLEAR_ERROR",
}

/**
 * type of the config context
 */
type TConfigContext = {
  themeMode: EThemeMode;
  error: string;
};

/**
 * config context consumer
 */
type TConfigContextConsumer = {
  state: TConfigContext;
  dispatch: React.Dispatch<{
    type: any;
    payload: any;
  }>;
};

const ConfigContext = createContext({} as TConfigContextConsumer);

const ConfigContextWrapper: React.FC<{
  children: React.ReactNode;
  defaultTheme: EThemeMode;
}> = ({ children, defaultTheme }) => {
  /**
   *
   * @param state - The current state of the ConfigContext.
   * @param action - The action to dispatch to the ConfigContext.
   * @returns - The new state of the ConfigContext.
   */
  const reducer = (
    state: TConfigContext,
    action: { type: keyof EConfigContextAction; payload: any }
  ) => {
    switch (action.type) {
      case EConfigContextAction.SET_THEME_MODE:
        return {
          ...state,
          themeMode: action.payload,
        };
      case EConfigContextAction.SET_ERROR:
        return {
          ...state,
          error: action.payload,
        };
      case EConfigContextAction.CLEAR_ERROR:
        return {
          ...state,
          error: "",
        };
      default:
        return state;
    }
  };

  /**
   * The reducer for the ConfigContext.
   */
  const [state, dispatch] = useReducer(reducer, {
    themeMode: defaultTheme,
    error: "",
  });

  return (
    <ConfigContext.Provider value={{ state, dispatch }}>
      {children}
    </ConfigContext.Provider>
  );
};

export default ConfigContextWrapper;

export function useConfigContext(): TConfigContextConsumer {
  return useContext(ConfigContext);
}
