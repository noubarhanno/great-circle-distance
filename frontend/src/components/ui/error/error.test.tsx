import React from "react";
import { render } from "@testing-library/react";
import Error from "./error";
import ContextWrapper from "context/config.context";
import * as Context from "context/config.context";

describe("Error test suite", () => {
  it("should match snopshot", () => {
    const { container } = render(
      <ContextWrapper defaultTheme="light">
        <Error />
      </ContextWrapper>
    );
    expect(container).toMatchSnapshot();
  });

  it("should match snopshot with error", async () => {
    jest.spyOn(Context, "useConfigContext").mockImplementation(() => ({
      state: { error: "error", themeMode: "light" },
      dispatch: jest.fn(),
    }));
    const { findByTestId } = render(
      <ContextWrapper defaultTheme="light">
        <Error />
      </ContextWrapper>
    );
    expect(findByTestId("snackbar-error")).toBeTruthy();
    expect(await findByTestId("snackbar-error")).toHaveTextContent("error");
  });
});
