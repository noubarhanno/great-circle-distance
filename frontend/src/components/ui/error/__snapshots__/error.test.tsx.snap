// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error test suite should match snopshot 1`] = `
<div>
  <div
    class="MuiBox-root css-42igfv"
    data-testid="error-component"
  />
</div>
`;
